umbral/fragments.py

```diff
@@ -109,7 +179,14 @@ def to_bytes(self):
         kfrag_id = self.bn_kfrag_id.to_bytes()
         eph_ni = self.point_eph_ni.to_bytes()
 
-        return e1 + v1 + kfrag_id + eph_ni
+        serialized_cfrag = e1 + v1 + kfrag_id + eph_ni
+
+        if self.proof is not None:
+            serialized_cfrag += self.proof.to_bytes()
```

Comment:
As well as here.

Reply:
Fixed

Reply:
mmm, no, it fails when `self.proof` is None. I'll revert it to the previous way.